conans/client/cache/remote_registry.py

```diff
@@ -40,316 +36,307 @@ def load_registry_txt(contents):
-            remotes, refs, prefs = self._load()
-            if remote_name not in remotes:
-                raise ConanException("Remote '%s' already exists" % new_remote_name)
-
-            if remote_name not in remotes:
-                raise ConanException("Remote '%s' not found in remotes" % remote_name)
-            new_remotes = OrderedDict()
-            for name, info in remotes.items():
-                name = name if name != remote_name else new_remote_name
-                new_remotes[name] = info
-            remotes = new_remotes
-            for k, v in refs.items():
-                if v == remote_name:
-                    refs[k] = new_remote_name
-            self._save(remotes, refs, prefs)
+        if new_remote_name in self._remotes:
+            raise ConanException("Remote '%s' already exists" % new_remote_name)
 
-    def define(self, remotes):
-        with fasteners.InterProcessLock(self._lockfile, logger=logger):
-            _, refs, prefs = self._load()
-            refs = {k: v for k, v in refs.items() if v in remotes}
-            self._save(remotes, refs, prefs)
-
-    def _add_update(self, remote_name, url, verify_ssl, exists_function, insert=None):
-        with fasteners.InterProcessLock(self._lockfile, logger=logger):
-            remotes, refs, prefs = self._load()
-            exists_function(remotes)
-            urls = {r[0]: name for name, r in remotes.items() if name != remote_name}
-            if url in urls:
-                raise ConanException("Remote '%s' already exists with same URL" % urls[url])
-            if insert is not None:
-                try:
-                    insert_index = int(insert)
-                except ValueError:
-                    raise ConanException("insert argument must be an integer")
-                remotes.pop(remote_name, None)  # Remove if exists (update)
-                remotes_list = list(remotes.items())
-                remotes_list.insert(insert_index, (remote_name, (url, verify_ssl)))
-                remotes = OrderedDict(remotes_list)
-            else:
-                remotes[remote_name] = (url, verify_ssl)
-            self._save(remotes, refs, prefs)
+        remote = self._remotes[remote_name]
+        new_remote = Remote(new_remote_name, remote.url, remote.verify_ssl)
+        self._remotes = OrderedDict([(new_remote_name, new_remote) if k == remote_name
+                                     else (k, v) for k, v in self._remotes.items()])
+
+    def get_remote(self, remote_name):
+        # Returns the remote defined by the name, or the default if is None
+        return self[remote_name] if remote_name is not None else self.default
 
     @property
     def default(self):
         try:
-            return self.list[0]
-        except IndexError:
-            raise NoRemoteAvailable("No default remote defined in %s" % self._filename)
+            return self._remotes[next(iter(self._remotes))]
```

Comment:
Could you put a comment line: "that is the way to get the first element of an OrderedDict" I kept looking at it for a minute.